components/AnnotationChart.tsx

```diff
@@ -0,0 +1,415 @@
+    legendPosition?: "bottom" | "left" | "top" | "right" | "center" | "chartArea" | undefined, // should include _DeepPartialObject<{ [scaleId: string]: number; }> ?
+    legendAlign?: "center" | "end" | "start" | undefined,
+}
+
+
+
+declare module 'chart.js' {
+   interface PluginOptionsByType<TType extends ChartType = ChartType> {
+        staticVerticalLine?: {
+            position: number,
+            metricsNumber: number,
+            color: string
+        },
+        dynamicVerticalLine?: {
+            metricsNumber: number,
+            color: string,
+            clickHandler: ()=> void, // return to this later
+            startIndex: number
+        }
+    } 
+}
+//Todo: migrate this to wherever IntervalErasing is declared in Workspace.  TS will infer the type of 'value' from here.
+type IntervalErasingValues = {
+    on: boolean,
+    startFrame: number | null,
+    videoID: string | null,
+    labels: string[]
+}
+
+export default function AnnotationChart({labels, width, height, staticVerticalLineColor, dynamicVerticalLineColor, omitXLabels, legendPosition, legendAlign}: AnnotationChartProps) {
+
+    const chartRef = useRef<ChartJS<"bar"> | null>(null);
+    const [dataToDisplay, setDataToDisplay] = useState<ChartData<'bar'>>({
+            labels: [],
+            datasets: []
+    });
+
+    const [options, setOptions] = useState<ChartOptions<'bar'>>({
+        plugins: {
+            zoom: {
+                pan: {
+                    enabled: true,
+                    mode: 'y',
+                    modifierKey: 'alt',
+                }
+            },
+        }
+    });
+    const [annotationForChart, setAnnotationForChart] = useState<AnnotationData>({frameNum:null, range: null, data: null});
+
+    const setFrameNumSignal = useStateSetters().setFrameNumSignal;
+    const frameNum = useStates().frameNum;
+    const totalFrameCount = useStates().videoMetaRef.current.totalFrameCount;
+    const annotationChartRange = useStates().annotationChartRange;
+    const videoId = useStates().videoId;
+    const intervalAnno = useStates().intervalAnno;
+    const categoryColors = useStates().categoryColors;
+    const videoMetaRef = useStates().videoMetaRef;
+    const setCancelIntervalAnno = useStateSetters().setCancelIntervalAnno;  
+    const updateAnnotationChart = useStates().updateAnnotationChart;
+    const setUpdateAnnotationChart = useStateSetters().setUpdateAnnotationChart;
+    const uploader = useStates().uploader;
+    const resetAnnotationChart = useStates().resetAnnotationChart;
+    const setResetAnnotationChart = useStateSetters().setResetAnnotationChart; 
+    const intervalErasing = useStates().intervalErasing;
+    const annotationRef = useStates().annotationRef;
+    const setGlobalInfo = useStateSetters().setGlobalInfo;
+
+    useEffect(() => {
+        if (uploader?.type && uploader?.file) {
+            setAnnotationForChart(oldValue => {return {frameNum: null, range: null, data: null}});
```

Comment:
There is no need to pass the oldvalue if you are not going to use it.

`setAnnotationForChart(() => {return {frameNum: null, range: null, data: null}});`